bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/Journal.java

```diff
@@ -832,6 +833,7 @@ public void scanJournal(long journalId, long journalPos, JournalScanner scanner)
                     scanner.process(journalVersion, offset, recBuff);
                 }
             }
+            return recLog.fc.position();
```

Comment:
Do we have any corner case ? Empty file, more then one file..

 it better to add a method in JournalChannel, isn't it?
We could then add specific unit tests about this crucial method

Reply:
According to my understanding, every bookie restart creates a new journal file on start of size 16 MB and the first 512 bytes contains header. The method also processes a single file only. IMHO, this is covering all the cases.